jokers/red_card.lua

```diff
@@ -0,0 +1,163 @@
+        pos = { x = 2, y = 2 },
+        message_colour = G.C.MONEY,
+    },
+    green = {
+        next = "red",
+        loc_key = folly_utils.prefix.joker .. "green_card",
+        pos = { x = 3, y = 2 },
+        message_colour = G.C.GREEN,
+    },
+}
+
+return {
+    key = "red_card",
+    name = "fj_red_card",
+    atlas = "folly_jokers",
+    pos = { x = 0, y = 2 },
+    config = {
+        extra = {
+            colour = "red",
+            triggers = 0,
+            mult = 3,
+            chips = 10,
+            money = 1,
+            reroll = 0.2,
+        },
+    },
+    calc_dollar_bonus = function(self, card)
+        if card.ability.extra.colour == "yellow" then
+            return card.ability.extra.money * card.ability.extra.triggers
+        end
+    end,
+    add_to_deck = function(self, card, from_debuff)
+        if card.ability.extra.colour == "green" then
+            self.update_reroll_cost(card)
+        end
+    end,
+    remove_from_deck = function(self, card, from_debuff)
+        local old = card.ability.extra.colour
+        card.ability.extra.colour = "red"
+        card.ability.extra.triggers = card.ability.extra.triggers + 1
+        self.update_reroll_cost(card)
+        card.ability.extra.triggers = card.ability.extra.triggers - 1
+        card.ability.extra.colour = old
+    end,
+    set_sprites = function(self, card, front)
+        if card.ability then
+            local colour = colours[card.ability.extra.colour]
+            card.children.center:set_sprite_pos(colour.pos)
+        end
+    end,
+    calculate = function(self, card, context)
+        if context.skipping_booster and not context.blueprint then
+            card.ability.extra.triggers = card.ability.extra.triggers + 1
+            G.E_MANAGER:add_event(Event({
+                func = function()
+                    self:switch_colour(card)
+                    --red by default
+                    local vars = { card.ability.extra.mult }
+                    if card.ability.extra.colour == "blue" then
+                        vars = { card.ability.extra.chips }
+                    end
+                    if card.ability.extra.colour == "yellow" then
+                        vars = { card.ability.extra.money }
+                    end
+                    if card.ability.extra.colour == "green" then
+                        vars = { card.ability.extra.reroll }
+                    end
+                    G.E_MANAGER:add_event(Event({
+                        func = function()
+                            card.children.center:set_sprite_pos(colours[card.ability.extra.colour].pos)
+                            return true
+                        end
+                    }))
+                    SMODS.calculate_effect({
+                        message = localize{
+                            type = "variable",
+                            key = "a_folly_"..card.ability.extra.colour.."_card",
+                            vars = vars },
```

Comment:
have you thought about it just showing the color its changing to instead?
might feel/look better